rpc_component/cli.py

```diff
@@ -242,17 +238,33 @@ def dependency(components_dir, **kwargs):
         )
 
 
-def metadata(components_dir, **kwargs):
-    metadata_dir = kwargs.pop("metadata_dir")
+def get_dependents(component_name, download_dir, textflag, releases_dir, components_dir):
+    dependents = []
+    components = c_lib.load_all_components(components_dir, releases_dir)
+    for component in components:
+        requirements = [{"name": component.name, "repo_url": component.repo_url, "sha": 0}] #0 = latest commit, or use
+        c_lib.download_requirements(requirements, download_dir)                             #component.releases[0].sha
+        dep_component_dir = os.path.join(download_dir, component.name)                      #for latest release
+        metadata = get_metadata(dep_component_dir, "get")
+        if metadata:
+            for dep in metadata['dependencies']:
+                if dep['name'] == component_name:
+                    if textflag:
```

Comment:
These functions have a clear flow in terms of inputs and outputs, which this seems to break. Elsewhere I have questioned the use of `textflag`, on the assumption it is useful I think it would be better to have the information returned by the function and leave the presentation to the end.